data/pins_sonata.xdc

```diff
@@ -92,6 +92,9 @@ set_property -dict { PACKAGE_PIN J13 IOSTANDARD LVCMOS33 } [get_ports {pmod0[4]}
 set_property -dict { PACKAGE_PIN E17 IOSTANDARD LVCMOS33 } [get_ports {pmod0[5]}]
 set_property -dict { PACKAGE_PIN D17 IOSTANDARD LVCMOS33 } [get_ports {pmod0[6]}]
 set_property -dict { PACKAGE_PIN K13 IOSTANDARD LVCMOS33 } [get_ports {pmod0[7]}]
+## Enable pull-ups because this I2C bus will often be undriven externally.
+set_property PULLTYPE PULLUP [get_ports {pmod0[0]}]
```

Comment:
Please do not add pull ups to the PMOD pins. They've got multiple uses only one of which is GPIO so a permanent pull up would be a surprising property. I've just had a look at the schematic for the digilent colour detector pmod: https://uk.farnell.com/digilent/410-348/pmod-colour-detector-dev-board/dp/2893222 and that has pinups on the board (which you can disable by removing some jumpers). It's pretty common for I2C breakouts to have their own pull-ups (all the Qwiic boards do for instance) and I think it's the r-pi hat that's the odd one out here where it expects I2C pull-ups on the main board (and the latest version of the Sonata board includes physical pin-ups as the FPGA ones are very weak).

Reply:
This makes sense, I have removed the pull ups now.